src/dcqc/file.py

```diff
@@ -308,6 +309,27 @@ def is_file_local(self) -> bool:
         """
         return self._local_path is not None
 
+    def already_staged(self) -> list[Path]:
```

Comment:
Just a high level question, but do you think this is something that should be tackled in fs-synapse?

Basically if a file has been staged, should fs-synapse know to use the existing staged file?

Reply:
maybe, having a function that just lets you check if it has already been staged in fs-synapse might be useful

Reply:
Hmmm, i had a closer look and apologies. The "stage" function is specific to this package. Would it make sense to incorporate the logic you've implemented in already_staged in there?

Reply:
I did that originally, but because it's a fair bit of logic and `stage` is already large I figured a separate function made sense.

Reply:
Makes sense to me: although this does seem like a helper function.